src/transforms/uglify.js

```diff
@@ -11,18 +13,39 @@ module.exports = async function(asset) {
     warnings: true,
     mangle: {
       toplevel: true
-    }
+    },
+    sourceMap: asset.options.sourceMaps ? {filename: asset.relativeName} : false
   };
 
   if (customConfig) {
     options = Object.assign(options, customConfig);
   }
 
   let result = minify(code, options);
+
   if (result.error) {
     throw result.error;
   }
 
+  if (result.map) {
+    result.map = await new SourceMap().addMap(JSON.parse(result.map));
```

Comment:
I think it's possible to get the map without converting to JSON first. https://github.com/mishoo/UglifyJS2/blob/master/lib/sourcemap.js#L94

Might be faster - maybe you could even access the original mappings from there so you don't need to decode it.